bootloader/src/pyi_utils.c

```diff
@@ -895,33 +927,51 @@ pyi_utils_create_child(const char *thisfile, const int argc, char *const argv[])
             VS("WARNING: Application is started by systemd socket,"
                "but we can't set proper LISTEN_PID on it.\n");
         }
-        execvp(thisfile, argv_pyi);
+        if (execvp(thisfile, argv_pyi) < 0) {
+            VS("Failed to exec: %s\n", strerror(errno));
+            return 1;
+        }
+        /* NOTREACHED */
     }
-    /* Parent code. */
-    else {
-        child_pid = pid;
 
-        /* Redirect termination signals received by parent to child process. */
-        signal(SIGINT, &_signal_handler);
-        signal(SIGKILL, &_signal_handler);
-        signal(SIGTERM, &_signal_handler);
-    }
+    /* From here to end-of-function is parent code (since the child exec'd). */
 
-    wait(&rc);
+    child_pid = pid;
+
+    /* Redirect all signals received by parent to child process. */
+    if (ignore_signals) {
+        VS("LOADER: Ignoring all signals in parent\n");
+    } else {
+        VS("LOADER: Registering signal handlers\n");
+    }
+    for (signum = 0; signum < num_signals; ++signum) {
+        if (signum == SIGCHLD || signum == SIGCLD) {
+            // don't mess with SIGCHLD/SIGCLD; it affects our ability
+            // to wait() for the child to exit
+            handlers[signum] = SIG_ERR;
```

Comment:
Assigning SIG_ERR here did irritate me, I first thought this is overwriting the SIGCHLD handler. Why not simply sett this the to Null, which perfectly states: no handler for this signal installed